lib/S3Plugin.js

```diff
@@ -105,8 +105,15 @@ class S3Plugin {
     }
 
     this.context = context;
-
-    this.baseFileUrl = `https://s3.${this.config.region}.amazonaws.com/${this.config.bucketName}`;
+    //Build the endpoint parameter if we only have the region
+    if (this.config.endpoint === 'undefined') {
```

Comment:
```suggestion
    if (typeof this.config.endpoint === 'undefined') {
```
I think you forgot the `typeof` operator ?